### DISCUSSION TITLE
findFirst inside an include

### DISCUSSION THREAD

**Discussion by AltumSonatur** [prisma/prisma#12415] [Q&A] [answered]
Hi, i need to query a User which has a one-to-many relation with Post like this :
```
model User {
  id          Int        @id @default(autoincrement())
  name    String  @unique
  posts    Post[]
}

model Post {
 id         Int        @id @default(autoincrement())
 title     String   @unique
 user    User
```
When i make a query like below, the expected response is a user with a nested array containing the last post associated to this user.
Query :
```
const response = await prisma.user.findUnique({
    where: {
       name: user.name
    },
    include: {
       posts: {
         orderBy: { id: 'desc' },
         take: 1,
      },
    },
  });
```
Response :
```
{
user: {
   id: 1
   name: "John Doe"
   posts: [
      {
         id: 9
         title: "Hello world"
      }
   ]
}
```
Instead of having an array with one entry (the last post), is there a way to obtain directly the last post (no array) like that :
```
{
user: {
   id: 1
   name: "John Doe"
   post: {
      id: 9
      title: "Hello world"
   }
}
```

Thanks for your help !

**Comment by austincrim** [comment, marked as answer]
Hey @AltumSonatur 👋 ,

This is not possible yet with Prisma. I'm curious, are you wanting this for convenience or some other use case?

**Reply by AltumSonatur** [reply]
Hi @austincrim thanks !
Yes it's a lot more convenient and easy to work on a frontend with
`data.user.post.title`
instead of
`data.user.posts[0].title`

**Reply by austincrim** [reply]
Makes sense! If you have an idea for how this feature could look in Prisma, feel free to open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!

**Reply by lucia-cufre** [reply]
Hi This feature would be really usefull, please consider

**Comment by gabrielemidulla** [comment]
Hi, I've published what seems to be a nice workaround, it works by editing the index.d.ts of the generated PrismaClient and handling 1:1 relations at runtime.

[prisma-one2one](https://github.com/gabrielemidulla/prisma-one2one)

**Reply by igoro00** [reply]
the repo has been deleted
